esphome/components/uart/uart_component_esp32_arduino.cpp

```diff
@@ -86,10 +86,20 @@ void ESP32ArduinoUARTComponent::setup() {
   is_default_tx = tx_pin_ == nullptr || tx_pin_->get_pin() == 1;
   is_default_rx = rx_pin_ == nullptr || rx_pin_->get_pin() == 3;
 #endif
-  if (is_default_tx && is_default_rx) {
+  static uint8_t next_uart_num = 0;
+  if (is_default_tx && is_default_rx && next_uart_num == 0) {
     this->hw_serial_ = &Serial;
+    next_uart_num++;
   } else {
-    static uint8_t next_uart_num = 1;
+#ifdef USE_LOGGER
+    // The logger doesn't use this UART component, instead it targets the UARTs
+    // directly (i.e. Serial/Serial0, Serial1, and Serial2). If the logger is
+    // enabled, skip the UART that it is configured to use.
+    if (logger::global_logger->get_baud_rate() > 0 && logger::global_logger->get_uart() == next_uart_num) {
+      next_uart_num++;
+    }
+#endif  // USE_LOGGER
+
```

Comment:
This requires `#include <driver/uart.h>` at the top

```suggestion

    if (next_uart_num >= UART_NUM_MAX) {
      ESP_LOGW(TAG, "Maximum number of UART components created already.");
      this->mark_failed();
      return;
    }
```

Reply:
Incorporated in the updated commit